piff/kolmogorov_model.py

```diff
@@ -0,0 +1,311 @@
+        """ Run lmfit.minimize with given lmfit.Parameters() and on given star data.
+
+        :param params: lmfit.Parameters() instance (holds initial guess and which params to let
+                       float or hold fixed).
+        :param star:   Star to fit.
+
+        :returns: lmfit.MinimizerResult instance containing fit results.
+        """
+        import lmfit
+        if logger:
+            import time
+            t0 = time.time()
+            logger.debug("Start lmfit minimize.")
+        results = lmfit.minimize(cls._lmfit_resid, params, args=(star,))
+        if logger:
+            logger.debug("End lmfit minimize.  Elapsed time: {0}".format(time.time() - t0))
+        return results
+
+    def lmfit(self, star, logger=None):
+        """Fit parameters of the given star using lmfit (Levenberg-Marquardt minimization
+        algorithm).
+
+        :param star:    A Star to fit.
+        :param logger:  A logger object for logging debug info. [default: None]
+
+        :returns: (flux, cenx, ceny, sigma, g1, g2, flag)
+        """
+        params = self._lmfit_params(star)
+        results = self._lmfit_minimize(params, star, logger=logger)
+        if logger:
+            import lmfit
+            logger.debug(lmfit.fit_report(results))
+        return results.params.valuesdict().values() + [0 if results.success else 1]
+
+    def fit(self, star, fastfit=False, logger=None):
+        """Fit the image either using HSM or lmfit.
+
+        If `fastfit` is True, then the galsim.hsm module will be used to estimate the parameters of
+        the Kolmogorov PSF from image moments.  If `fastfit` is False, then the Levenberg-Marquardt
+        minimization algorithm as implemented by lmfit will be used instead.  The latter should
+        generally be more accurate, but slower due to the need to iteratively propose model
+        improvements.
+
+        :param star:    A Star to fit.
+        :param fastfit: Use fast HSM moments to fit? [default: False]
+        :param logger:  A logger object for logging debug info. [default: None]
+
+        :returns: a new Star with the fitted parameters in star.fit
+        """
+        if self._fastfit or fastfit:
+            flux, cenu, cenv, fwhm, g1, g2, flag = self.hsmfit(star)
+        else:
+            flux, cenu, cenv, fwhm, g1, g2, flag = self.lmfit(star, logger=logger)
+
+        # Make a StarFit object with these parameters
+        if self._force_model_center:
+            params = np.array([ fwhm, g1, g2 ])
+            center = (cenu, cenv)
+        else:
+            params = np.array([ cenu, cenv, fwhm, g1, g2 ])
+            center = (0.0, 0.0)
+
+        # Also need to compute chisq
+        prof = self.getProfile(params) * flux
+        model_image = star.image.copy()
+        prof.shift(center).drawImage(model_image, method='no_pixel',
+                                     offset=(star.image_pos - model_image.trueCenter()))
```

Comment:
I trust that you figured this out correctly, but since I get really quite confused about all this, I want to double check: in gaussian_model we compute the profile without the WCS, and compute the offset as

    galsim.PositionD(cenx, ceny) - star.image.trueCenter()

Here, you compute a shift in wcs coordinates (cenu = moment_center_u - image_pos_u) and then apply another offset in the drawing. This probably illustrates my ignorance of the galsim coordinate systems, but why not just group both into one thing, ie everything in prof.shift, or in the offset kwarg?